File: src/main/java/com/syfe/personalfinance/seed/CategorySeeder.java
package com.syfe.personalfinance.seed;

import com.syfe.personalfinance.entity.Category;
import com.syfe.personalfinance.enums.CategoryType;
import com.syfe.personalfinance.repository.CategoryRepository;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

import java.util.Arrays;
import java.util.List;

@Component
@Slf4j
public class CategorySeeder implements CommandLineRunner {

    private final CategoryRepository categoryRepository;

    public CategorySeeder(CategoryRepository categoryRepository) {
        this.categoryRepository = categoryRepository;
    }

    @Override
    public void run(String... args) {
        log.info("Bootstrapping default global categories...");

        List<Category> defaultCategories = Arrays.asList(
            Category.builder().name("Salary").type(CategoryType.INCOME).isDefault(true).build(),
            Category.builder().name("Food").type(CategoryType.EXPENSE).isDefault(true).build(),
            Category.builder().name("Rent").type(CategoryType.EXPENSE).isDefault(true).build(),
            Category.builder().name("Transportation").type(CategoryType.EXPENSE).isDefault(true).build(),
            Category.builder().name("Entertainment").type(CategoryType.EXPENSE).isDefault(true).build(),
            Category.builder().name("Healthcare").type(CategoryType.EXPENSE).isDefault(true).build(),
            Category.builder().name("Utilities").type(CategoryType.EXPENSE).isDefault(true).build()
        );

        for (Category defaultCat : defaultCategories) {
            if (!categoryRepository.existsByNameIgnoreCaseAndUserIsNull(defaultCat.getName())) {
                categoryRepository.save(defaultCat);
                log.debug("Seeded default category: {}", defaultCat.getName());
            }
        }

        log.info("Default global categories bootstrap completed successfully.");
    }
}

File: src/main/java/com/syfe/personalfinance/seed/CategorySeeder.kt
package com.syfe.personalfinance.seed

import com.syfe.personalfinance.entity.Category
import com.syfe.personalfinance.enums.CategoryType
import com.syfe.personalfinance.repository.CategoryRepository
import org.slf4j.LoggerFactory
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component

@Component
class CategorySeeder(
    private val categoryRepository: CategoryRepository
) : CommandLineRunner {

    private val log = LoggerFactory.getLogger(CategorySeeder::class.java)

    override fun run(vararg args: String?) {
        log.info("Bootstrapping default global categories...")

        val defaultCategories = listOf(
            Category(name = "Salary", type = CategoryType.INCOME, isDefault = true),
            Category(name = "Food", type = CategoryType.EXPENSE, isDefault = true),
            Category(name = "Rent", type = CategoryType.EXPENSE, isDefault = true),
            Category(name = "Transportation", type = CategoryType.EXPENSE, isDefault = true),
            Category(name = "Entertainment", type = CategoryType.EXPENSE, isDefault = true),
            Category(name = "Healthcare", type = CategoryType.EXPENSE, isDefault = true),
            Category(name = "Utilities", type = CategoryType.EXPENSE, isDefault = true)
        )

        for (defaultCat in defaultCategories) {
            if (!categoryRepository.existsByNameIgnoreCaseAndUserIsNull(defaultCat.name)) {
                categoryRepository.save(defaultCat)
                log.debug("Seeded default category: {}", defaultCat.name)
            }
        }

        log.info("Default global categories bootstrap completed successfully.")
    }
}
